packages/nc-gui/components/smartsheet/Grid.vue

```diff
@@ -203,7 +202,7 @@ const makeEditable = (row: Row, col: ColumnType) => {
 /** handle keypress events */
 const onKeyDown = async (e: KeyboardEvent) => {
   if (e.key === 'Alt') {
-    disableUrlOverlay.value = true
+    disableOverlay.value = true
```

Comment:
I am new to Vue, but wouldn't mutating a composable value like this be considered an anti-pattern? Makes it very hard to track where changes are coming from.

Reply:
Not really.

How would your implementation be easier to track? It does the exact same for the cost of more code and variables.

Reply:
I think the concept of composables needs a bit of clarification.
Composables aren't a rigid thing, but the core concept is obviously reusability of code.

Let's say we want to implement a simple toggle

```vue
<script lang="ts" setup>
const state = ref(false)

const toggleState = (nextState?: boolean) => (state.value = nextState ?? !state.value)
</script>

<template>
	<button @click="toggleState">Toggle</button>
</template>
```

In this example we can easily extract the functionality provided inside the setup function and make it reusable across any component for multiple purposes.

```ts
// useToggle.ts

function useToggle(initialState?: boolean) {
	const state = ref(initialState ?? false)
	
	const toggleState = (nextState?: boolean) => (state.value = nextState ?? !state.value)

	return [state, toggleState]
}
```

And now we can simply use our composable inside the component and save us the trouble of initializing refs etc. for the purpose of using a toggle

```vue
<script lang="ts" setup>
import { useToggle } from './useToggle'

const [state, toggleState] = useToggle()
</script>

<template>
	<button @click="toggleState">Toggle</button>
</template>
```

Simple enough.
`useToggle` will always initialize a new scope - meaning, every time you call it there will be a new ref for `state` and `toggleState` will always reference that `state` ref.

`useCellUrlConfig` does not do this - that's why it's using `useInjectionState`. 
Meaning, every time you call `useCellUrlConfig` there will only be a *single*  instance of it's scope that is available - never multiple.
Even if you called the `provide` function multiple times, it would only overwrite the state and not add a second one that could be used.
Basically like a Vuex or Pinia store. It's a single instance of a store to be used across a component tree, beginning from the point it was provided.

You're not wrong that it becomes harder to track where something was provided and possibly mutated, but using explicit `inject` and `provide` calls doesn't really remedy that situation, it just uses a different syntax to do the same thing 😄 

I also believe that, in the current state of the app, there is a lot of improvement in terms of setups and initialization so it becomes less troublesome to track values, which is something that will be worked on of course.

I hope this clears up why I decided to include the variables inside a single location instead of spreading it out into multiple locations.

Reply:
Thank you for the explanation

Reply:
If you're coming from a React background, don't confuse composables with React Hooks.

React Hooks are *similar* but composables are a lot more flexible and can basically be any type of functionality you want to re-use unlike React Hooks which, in most cases, *have* to be called inside a Functional Components' body.